R/verb-expand.R

```diff
@@ -0,0 +1,44 @@
+#' @importFrom tidyr expand
+#' @export
+expand.tbl_lazy <- function(data, ..., .name_repair = "check_unique") {
+  # TODO sort output?
+  # TODO wait for bugfix: distinct() ignores groups
+  # https://github.com/tidyverse/dbplyr/issues/535
+  dots <- quos_auto_name(quos(...))
+  dots <- purrr::discard(dots, rlang::quo_is_null)
+  names(dots) <- vctrs::vec_as_names(names(dots), repair = .name_repair)
+
+  if (is_empty(dots)) {
+    # TODO what should be returned here?
+    abort("No columns selected")
+  }
+
+  distinct_tables <- purrr::imap(
+    dots,
+    ~ {
+      # ugly hack to deal with `nesting()`
+      if (rlang::quo_is_call(.x) && rlang::call_name(.x) == "nesting") {
```

Comment:
Can simplify this by using the `name`  arg to `quo_is_call()`. Also note the rlang namespace is already imported, so you don't need `rlang::`.

I think the hack is fine; doing it 100% correctly is going to be a lot of work (and it's not even clear to me how you would do it)